### PR TITLE
DMP-747 Store Daily List XML

### PR DESCRIPTION
**Pull request by cmagowan-hmcts**
https://tools.hmcts.net/jira/browse/DMP-747

### CODE REVIEW

File: src/main/resources/openapi/dailyList.yaml [hunk 1]
```diff
@@ -27,11 +27,88 @@ paths:
...
+          name: json_document
+          schema:
+            $ref: '#/components/schemas/PostDailyListJson'
+          description: "JSON representation of the 'document' received in the addDocument request.<p>
+            **Conditional mandatory** either this or xml_document needs to be provided, or both."
+
       responses:
-        '201':
+        '200':
           description: Created
```
Comment (HMCTSDeveloper) [inline src/main/resources/openapi/dailyList.yaml:71]: This should be OK

File: src/main/java/uk/gov/hmcts/darts/dailylist/controller/DailyListController.java [hunk 2]
```diff
@@ -46,32 +51,83 @@ public class DailyListController implements DailyListsApi {
     @Autowired
     private DailyListProcessor processor;
 
+    @Override
```
Comment (HMCTSDeveloper) [inline src/main/java/uk/gov/hmcts/darts/dailylist/controller/DailyListController.java:54]: Can we remove these annotations and leave them in the supper class. it makes looking at the rest controller hard than it needs to be.

**Review by HMCTSDeveloper — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/darts/dailylist/validation/DailyListPostValidator.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+@UtilityClass
+public class DailyListPostValidator {
+
+    public void validate(DailyListPostRequest request) {
+        if (request.getDailyListJson() == null) {
+            if (StringUtils.isBlank(request.getDailyListXml())) {
+                throw new DartsApiException(DailyListError.XML_OR_JSON_NEEDS_TO_BE_PROVIDED);
+            }
+            if (StringUtils.isNotBlank(request.getDailyListXml())
```
[the remark this answers is not in the document]
Comment (cmagowan-hmcts) [reply src/main/java/uk/gov/hmcts/darts/dailylist/validation/DailyListPostValidator.java]: done

File: src/main/resources/openapi/dailyList.yaml (hunk 1 above)
Comment (cmagowan-hmcts) [reply src/main/resources/openapi/dailyList.yaml]: done

File: src/main/java/uk/gov/hmcts/darts/dailylist/controller/DailyListController.java (hunk 2 above)
Comment (cmagowan-hmcts) [reply src/main/java/uk/gov/hmcts/darts/dailylist/controller/DailyListController.java:54]: done

**Review by HMCTSDeveloper — APPROVED** [review]

**Pull request merged**